src/components/AddComment.jsx

```diff
@@ -13,12 +13,19 @@ const AddComment = ({ article_id, comments, setComments }) => {
     const optimisticComment = {
         comment_id: -1, votes: 0, created_at: "now",author: userData.username, body: commentInput,   article_id: article_id
     }
-    setComments([optimisticComment, ...comments]);
+    event.target.disabled = true
+    
 
-    postComment(article_id, commentInput, userData.username).catch(() => {
-        setError("Woopsie! There was an error commenting! Please try again later...")
+    postComment(article_id, commentInput, userData.username).then(() => {
+      setComments([optimisticComment, ...comments])
```

Comment:
This being inside the then block means it is no long optimistically rendered.
setting the comment input back to "" makes sense in the then block so it only resets when the post has actually worked.